development/ts-migration-dashboard/scripts/build-module-partitions.ts

```diff
@@ -0,0 +1,336 @@
+import fs from 'fs';
```

Comment:
This file looks at the current state of the repo to determine all of the files that have been converted to TypeScript, filtered by the list of original files to convert, and produces the data structure the dashboard uses to render the UI. To render the various levels displayed on the UI (explained in the ["What is this?" section](https://metamask.github.io/metamask-extension-ts-migration-dashboard/)) we use the [`madge`](https://github.com/pahen/madge) package to iterate through the dependencies and subdependencies of a set of starting files.

As the code here is expensive to run, it doesn't get run when you load the dashboard, but rather during the build process. More specifically, it's run once when you run `yarn ts-migration:dashboard:build` or `yarn ts-migration:dashboard:watch`. The build script writes the output of this file to a temporary JSON file, and that JSON file is then read by the React app.